File: Editorial/0137_Single_Number_II/C/2_Hash_Map.c
typedef struct {
    int num;
    int freq;
    UT_hash_handle hh;
} NumFreq;
int singleNumber(int* nums, int numsSize) {
    NumFreq* freq = NULL;
    for (int i = 0; i < numsSize; i++) {
        NumFreq* s;
        HASH_FIND_INT(freq, &nums[i], s);
        if (s == NULL) {
            s = (NumFreq*)malloc(sizeof(NumFreq));
            s->num = nums[i];
            s->freq = 1;
            HASH_ADD_INT(freq, num, s);
        } else {
            s->freq++;
        }
    }
    int loner = 0;
    for (NumFreq* s = freq; s != NULL; s = s->hh.next) {
        if (s->freq == 1) {
            loner = s->num;
            break;
        }
    }
    HASH_CLEAR(hh, freq);
    return loner;
}

File: Editorial/0101-0200/0137_Single_Number_II/Go/2_Hash_Map.go
func singleNumber(nums []int) int {
    freq := make(map[int]int)
    for _, num := range nums {
        freq[num]++
    }
    var loner int
    for num, count := range freq {
        if count == 1 {
            loner = num
            break
        }
    }
    return loner
}
